From 27d1acfc687e44a4c9f7d8788f635513b99db12e Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Thu, 8 Jul 2021 22:14:56 +0200
Subject: [PATCH] Don't let "tar" manage compression (s-x86)

It may end up calling "zcat", which is not allowed in the building
sandbox. As "gzip" is present int prebuilts, just change the command to
use "gzip" separately.
---
 wl/Android.mk | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/wl/Android.mk b/wl/Android.mk
index da51e46..fa5e319 100644
--- a/wl/Android.mk
+++ b/wl/Android.mk
@@ -34,16 +34,12 @@ WL_PATCHES := \
 	012-linux590.patch \
 	013-linux5100.patch \
 
-ifndef TAR_PATH
-  TAR_PATH = /usr/bin/tar
-endif
-
 $(WL_SRC):
 	@echo Downloading $(@F)...
 	$(hide) mkdir -p $(@D) && /usr/bin/curl -k https://docs.broadcom.com/docs-and-downloads/docs/linux_sta/$(@F) > $@
 
 $(WL_LIB): $(WL_SRC) $(addprefix $(LOCAL_PATH)/,$(WL_PATCHES))
-	$(hide) $(TAR_PATH) zxf $< -C $(@D) --overwrite -m && \
+	$(hide) gzip -d -c $< | tar xf - -C $(@D) --overwrite -m && \
 		rm -rf $@ && mv $(@D)/lib $@ && touch $@ && \
 		cat $(filter %.patch,$^) | patch -p1 -d $(@D)
 
-- 
2.25.1

